Parcial 1 Minería de datos

In [8]:
!pip install dfply

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 612 kB 5.3 MB/s 


In [9]:
import pandas as pd
import numpy as np
import csv
from sklearn import datasets
from dfply import *
import numpy as np

In [2]:
pd.set_option('display.max_columns', None) #Mostrar todas las columnas

In [3]:
df = pd.read_csv(
    'https://raw.githubusercontent.com/prof-tatiana-gutierrez/DatosMD/master/student-mat.csv',
    sep = ";")
df.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher   
1     GP   F   17       U     GT3       T     1     1  at_home     other   
2     GP   F   15       U     LE3       T     1     1  at_home     other   
3     GP   F   15       U     GT3       T     4     2   health  services   
4     GP   F   16       U     GT3       T     3     3    other     other   

   reason guardian  traveltime  studytime  failures schoolsup famsup paid  \
0  course   mother           2          2         0       yes     no   no   
1  course   father           1          2         0        no    yes   no   
2   other   mother           1          2         3       yes     no  yes   
3    home   mother           1          3         0        no    yes  yes   
4    home   father           1          2         0        no    yes  yes   

  activities nursery higher internet romantic  famrel  freetime  goout  Dalc  \
0         no     yes    yes       no       no       4         3      4     1   
1         no      no    yes      yes       no       5         3      3     1   
2         no     yes    yes      yes       no       4         3      2     2   
3        yes     yes    yes      yes      yes       3         2      2     1   
4         no     yes    yes       no       no       4         3      2     1   

   Walc  health  absences  G1  G2  G3  
0     1       3         6   5   6   6  
1     1       3         4   5   5   6  
2     3       3        10   7   8  10  
3     1       5         2  15  14  15  
4     2       5         4   6  10  10

# Primer punto

In [60]:
df.isnull().sum()

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

Respuesta= No Hay valores vacios o nulos para alguna delas variables del data set, la base esta completa

# Segundo punto

In [65]:
df['address'].value_counts()

U    307
R     88
Name: address, dtype: int64

Hay 307 estudiantes que viven en zona urbana y 88 estudiantes viven en zona rural, con 395 estudiantes en total de la base

In [66]:
df2 = (df >>
 mask(X.address =='R', X.sex=='M'))
df2['sex'].value_counts()

M    44
Name: sex, dtype: int64

De los estudiantes que viven en zona rural, 44 son hombres

# Tercer punto

In [34]:
df3 = (df >>
 mask(X.address =='R', X.absences >5))
df3['absences'].value_counts().sum()

38

Hay 38 estudiantes que viven en la zona rural los cuales tienen mas de 5 fallas

In [36]:
df3['sex'].value_counts()

F    20
M    18
Name: sex, dtype: int64

18 de los estudiantes que viven en una zona rural y tienen mas de 5 fallas son hombres

# Cuarto punto

In [44]:
df4 = (df >>
 select(X.school, X.G3, X.internet)) # #Tomando las columnas que se quieren
df4.head()

school  G3 internet
0     GP   6       no
1     GP   6      yes
2     GP  10      yes
3     GP  15      yes
4     GP  10       no

In [55]:
pd.pivot_table(df4, index=['school','internet'], aggfunc=np.mean)

G3
school internet           
GP     no         9.673077
       yes       10.632997
MS     no         8.428571
       yes       10.468750

- Cuando se tiene internet, la media de las notas de matematicas en el colegio GP es de 10.632, y la del colegio MS es de 10.
468
- Cuando no se tiene internet, la media de las notas de matematicas en el colegio GP es de 9.673, y la del comlgio MS es de 8.428
- Hay diferencias numericas entre los promedos de notas de matematicas cuando se tiene o no se tiene internet en ambos colegios

# Quinto punto

In [23]:
df5 = (df >>
 mutate(
   Calificacion_total = X.G1*0.3+X.G2*0.3+X.G3*0.4, # Nueva variable
 ))
df5.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher   
1     GP   F   17       U     GT3       T     1     1  at_home     other   
2     GP   F   15       U     LE3       T     1     1  at_home     other   
3     GP   F   15       U     GT3       T     4     2   health  services   
4     GP   F   16       U     GT3       T     3     3    other     other   

   reason guardian  traveltime  studytime  failures schoolsup famsup paid  \
0  course   mother           2          2         0       yes     no   no   
1  course   father           1          2         0        no    yes   no   
2   other   mother           1          2         3       yes     no  yes   
3    home   mother           1          3         0        no    yes  yes   
4    home   father           1          2         0        no    yes  yes   

  activities nursery higher internet romantic  famrel  freetime  goout  Dalc  \
0         no     yes    yes       no       no       4         3      4     1   
1         no      no    yes      yes       no       5         3      3     1   
2         no     yes    yes      yes       no       4         3      2     2   
3        yes     yes    yes      yes      yes       3         2      2     1   
4         no     yes    yes       no       no       4         3      2     1   

   Walc  health  absences  G1  G2  G3  Calificacion_total  
0     1       3         6   5   6   6                 5.7  
1     1       3         4   5   5   6                 5.4  
2     3       3        10   7   8  10                 8.5  
3     1       5         2  15  14  15                14.7  
4     2       5         4   6  10  10                 8.8

In [24]:
df5['Calificacion_total'].describe()

count    395.000000
mean      10.652911
std        3.769058
min        1.200000
25%        8.300000
50%       10.700000
75%       13.300000
max       19.400000
Name: Calificacion_total, dtype: float64

Para la calificacion total de cada uno de los estudiantes
- La media es 10.65
- La mediana es 10.7
- El valor minimo obtenido por el/los estudiantes en la calificacion total fue 1.2
- El valor maximo obtenido por el/los estudiante/es en la calificacion total fue 19.4

# Sexto punto

Con los datos como estan realizaria una regresion, ya que considero que interesaria saber el como van a ser las notas de los estudiantes en el corte final (G3) basandose en toda la informacion demografica de los estudiantes, y las notas de los estudiantes en el corte final (G3) es una variable continua. Si dependiendo de un rango de notas por ejemplo 0-10=malo 0-15=regular y 15-20=exelente existiera esa variable en la base, trabajaria con esa y haria una clasificacion, pero al no estar, creo que es mejor la regresion